cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/ResourceDistributionGoal.java

```diff
@@ -260,6 +260,10 @@ protected void updateGoalState(ClusterModel clusterModel, OptimizationOptions op
     // Log broker Ids over balancing limit.
     // While proposals exclude the excludedTopics, the balance still considers utilization of the excludedTopic replicas.
```

Comment:
If all brokers are under low utilization threshold, shouldn't we update the following in `initGoalState` function to make sure that balancing limits are set properly?

```
  private double _balanceUpperThreshold;
  private double _balanceLowerThreshold;
```

Reply:
After some offline discussion, we decided to only look at the overall cluster resource utilization over brokers that are not excluded for replica move instead of whether all brokers are under the low utilization threshold. The reason is that suppose we have a cluster of 3 broker A, B, and C where A and B have no resource utilization and C has some resource utilization that is slightly over the low utilization threshold. In this case, we adjust the lower/upper thresholds basing on the low utilization threshold instead of using the thresholds calculated from the original resource utilization percentage given in the config. This way, the rebalance might involve fewer operations.

In general, the gist is to set more lenient and reasonable values to the balance upper/lower threshold parameters instead of avoiding rebalance completely.